lib/vdyp-forward/src/main/java/ca/bc/gov/nrs/vdyp/forward/ForwardDataStreamReader.java

```diff
@@ -231,20 +260,32 @@ private void calculateSpeciesCoverage(VdypSpecies s, Map<UtilizationClass, VdypU
 	protected void applyGroups(BecDefinition bec, String genus, VdypSpecies species) {
 		// Look up Volume group, Decay Group, and Breakage group for each species.
 
-		var volumeGroupMap = resolvedControlMap.getVolumeEquationGroups();
-		var decayGroupMap = resolvedControlMap.getDecayEquationGroups();
-		var breakageGroupMap = resolvedControlMap.getBreakageEquationGroups();
+		try {
+			// VGRPFIND
+			var volumeGroupMap = resolvedControlMap.getVolumeEquationGroups();
+			var volumeGroup = volumeGroupMap.get(genus, bec.getVolumeBec().getAlias());
+			species.setVolumeGroup(volumeGroup);
+		} catch (NoSuchElementException e) {
+			// group will remain undefined
+		}
 
-		// VGRPFIND
-		var volumeGroup = volumeGroupMap.get(genus, bec.getVolumeBec().getAlias());
-		// DGRPFIND
-		var decayGroup = decayGroupMap.get(genus, bec.getDecayBec().getAlias());
-		// BGRPFIND (Breakage uses decay BEC)
-		var breakageGroup = breakageGroupMap.get(genus, bec.getDecayBec().getAlias());
+		try {
+			// DGRPFIND
```

Comment:
Wouldn't it make more sense to have these methods on the resolved control map return an optional instead of having them throw an exception you catch here?

Reply:
I wanted to avoid Optionals being returned from ResolvedControlMap since during normal operation all these values will be present. It's only in tests that they're not, and putting it in the interface implies otherwise.